pkg/periodic/periodic.go

```diff
@@ -25,15 +25,35 @@ type Tick struct {
 	Time    time.Time
 }
 
+// Immediate starts the periodic task immediately instead of after the first tick.
+func Immediate() Option {
+	return optionFunc(func(p *periodic) {
+		p.immediate = true
+	})
+}
+
 // OnStop configures a callback that is executed when a periodic task is stopped or canceled.
 func OnStop(f func(Tick)) Option {
 	return optionFunc(func(p *periodic) {
 		p.onStop = f
 	})
 }
 
+// Stable puts the periodic task into stable mode,
+// which guarantees to always wait for the specified interval
+// between task executions.
```

Comment:
Wouldn't the overall PR work just as well without this additional flag? Also I find naming this `stable` confusing, is that terminology used anywhere else?

Reply:
Would work without it too. But the lower the interval, the higher the chance that you'll have consecutive runs that are under the wait time. And I wanted to keep the interval constant. I just chose `stable` like that. We could also go with `constant` or something like that. But it's also fine to just remove it.

Reply:
When the feature is activated initially, this counts as a single run anyways, doesn't it?

So I think the exact details here only affect continuous operation anyways. Averaged over a longer time, you should end up with the same fraction of time the feature is running with both variants. Let's say that fraction is 50% (probably highly exaggerated) and the interval is 1 hours, this would either make it run for 30 minutes every hour or for 1 hour every 2 hours.

I don't see a meaningful difference, so I'd go with the simpler option (that would be using the existing periodically implementation as-is).

Things might look a bit different if there was more control on when exactly the cleanup is running, so that you could move it to the night or weekends. (Please don't take this as a request to implement this, it would probably be overkill.)

Reply:
Removed.